parlai/agents/transformer/polyencoder.py

```diff
@@ -139,17 +150,24 @@ def score_candidates(self, batch, cand_vecs, cand_encs=None):
                 cand_rep = cand_encs.expand(bsz, cand_encs.size(1), -1)
         # bsz x num cands x seq len
         elif len(cand_vecs.shape) == 3:
-            _, _, cand_rep = self.model(cand_tokens=cand_vecs)
+            _, _, _, cand_rep = self.model(cand_tokens=cand_vecs)
         # bsz x seq len (if batch cands) or num_cands x seq len (if fixed cands)
         elif len(cand_vecs.shape) == 2:
-            _, _, cand_rep = self.model(cand_tokens=cand_vecs.unsqueeze(1))
+            _, _, _, cand_rep = self.model(cand_tokens=cand_vecs.unsqueeze(1))
             num_cands = cand_rep.size(0)  # will be bsz if using batch cands
             cand_rep = cand_rep.expand(num_cands, bsz, -1).transpose(0, 1).contiguous()
+
         scores = self.model(
-            ctxt_rep=ctxt_rep, ctxt_rep_mask=ctxt_rep_mask, cand_rep=cand_rep
+            ctxt_rep=ctxt_rep, ctxt_rep_mask=ctxt_rep_mask, cand_rep=cand_rep, ctxt_pos=ctxt_pos
         )
         return scores
 
+    def load_state_dict(self, state_dict):
+        """Override to account for codes"""
+        if self.model.type == 'codes' and 'codes' not in state_dict:
+            state_dict['codes'] = self.model.codes
```

Comment:
interesting solution, we could just do a `load_state_dict(..., strict=False)` but that might not capture other errors/keys missing

Reply:
I thought about that too, but I agree that it's better that we fail loudly when we pass in incorrect state dicts